http/src/v0/block.rs

```diff
@@ -221,7 +226,13 @@ async fn stat_query<T: IpfsTypes>(
     query: StatQuery,
 ) -> Result<impl Reply, Rejection> {
     let cid: Cid = query.arg.parse().map_err(StringError::from)?;
-    let block = ipfs.get_block(&cid).await.map_err(StringError::from)?;
+    let block = ipfs
+        .get_block(&cid)
+        .maybe_timeout(query.timeout.map(StringSerialized::into_inner))
+        .await
+        .map_err(StringError::from)?
+        .map_err(StringError::from)?;
```

Comment:
the
```
.maybe_timeout(query.timeout.map(StringSerialized::into_inner))
        .await
        .map_err(StringError::from)?
        .map_err(StringError::from)?
```
pattern seems pretty popular, perhaps it's a nice candidate for some macroization?

Reply:
I am pretty sure soon there'll be some required strings for interface on timeouts so ... Not so fond of prettying this up. Not that this is too readable either.

Alternatives I see:

 * define a custom crate::v0::support::Elapsed, make it a `warp::Rejection` or convertable, can't remember
 * ...?

The first alt would allow just

```diff
-.map_err(StringError::from)?
+?
```

Reply:
Yeah lets leave this ugly and revisit when we can return errors from streaming bodies maybe?